File: packages/restapi/src/lib/chat/latestMessage.ts
import Constants from '../constants';
import { AccountEnvOptionsType } from '../types';
import { IPGPHelper, PGPHelper } from './helpers';
import { history } from './historicalMessages';

export interface LatestMessagesOptionsType extends AccountEnvOptionsType {
  threadhash: string;
  toDecrypt?: boolean;
  pgpPrivateKey?: string;
}

/**
 * Get the latest chat message
 */

export const latest = async (options: LatestMessagesOptionsType) => {
  return await latestCore(options, PGPHelper)
}
export const latestCore = async (options: LatestMessagesOptionsType, pgpHelper: IPGPHelper) => {
  const {
    threadhash,
    pgpPrivateKey = '',
    account,
    toDecrypt = false,
    env = Constants.ENV.PROD,
  } = options || {};

  return history({
    threadhash,
    toDecrypt,
    limit: 1,
    pgpPrivateKey,
    account,
    env,
  });
};

File: packages/push-api-dart/lib/src/chat/src/latest_message.dart
import 'package:push_api_dart/push_api_dart.dart';

///Get the latest chat message
Future<List<IMessageIPFS>> latest({
  required String threadhash,
  required String account,
  int limit = FetchLimit.DEFAULT,
  required String pgpPrivateKey,
  bool toDecrypt = false,
}) async {
  return history(
    threadhash: threadhash,
    account: account,
    pgpPrivateKey: pgpPrivateKey,
    limit: 1,
    toDecrypt: toDecrypt,
  );
}
